application/views/main/install/requirement_check.php

```diff
@@ -171,23 +171,39 @@
 
 <?php if ($error > 0): ?>
 <div>
-	<p>Unfortunately, your system does not meet the minimum requirements to run Kalkun. Please update your system to meet the above requirements and refresh this page.</p>
+	<p>Unfortunately, your system does not meet the minimum requirements to run Kalkun. Please update your system to meet the above requirements. Then click on button to check again.</p>
+	<div align="center">
+		<?php
+	echo form_open('install/requirement_check');
+	echo form_hidden('idiom', $idiom);
+	echo '<input type="submit" name="submit" value="'.tr('Check again').'" class="button" />';
+	echo form_close();
+?>
+	</div>
+	<p>&nbsp;</p>
+	<?php
+	echo form_open('install', 'style="display:inline"');
+	echo form_hidden('idiom', $idiom);
+	echo '<input type="submit" name="submit" value="‹ '.tr('Previous').'" class="button" />';
+	echo form_close();
+?>
 </div>
 
 <?php else: ?>
 <div>
 	<p>Your system is compatible with Kalkun.</p>
+	<p>&nbsp;</p>
 	<p>
 		<?php
 	echo form_open('install', 'style="display:inline"');
 	echo form_hidden('idiom', $idiom);
-	echo form_submit('submit', '‹ '.tr('Previous'), 'class="button"');
+	echo '<input type="submit" name="submit" value="‹ '.tr('Previous').'" class="button" />';
```

Comment:
Any reason to switch from `form_submit` to bare HTML tags?

Reply:
yes, `form_input` does something like htmlentities. So it is done twice which results in double escape
& -> &amp; -> &amp;amp;

Reply:
In that case can't we use tr_raw which you created?

Reply:
preferably not. Not sure if form_submit of ci uses ENT_QUOTES while escaping.

Reply:
Uses html_escape https://github.com/bcit-ci/CodeIgniter/blob/develop/system/helpers/form_helper.php

What do you think?

Reply:
Let's use `form_submit` only then. That way we benefit from `ENT_QUOTES` along with the additional handlings

Reply:
I find it always a bit dangerous to use tr_raw(). A simple copy paste and you have an unescaped label.
Still not interested to have it this way?

Reply:
I just don't want to mix both HTML `<input...` and then `form_*`, because any issue can happen in both. So IMO we should choose one variant.

Reply:
Ok, it's changed.